advanced_source/cpp_export.rst

```diff
@@ -0,0 +1,366 @@
+- The ``include/`` folder contains header files your program will need to include,
+- The ``share/`` folder contains the necessary CMake configuration to enable the simple `find_package(Torch)` command above.
+
+The last step is building the application. For this, assume our example
+directory is layed out like this:
+
+.. code-block:: sh
+
+  example-app/
+    CMakeLists.txt
+    example-app.cpp
+
+We can now run the following commands to build the application from within the
+``example-app/`` folder:
+
+.. code-block:: sh
+
+  mkdir build
+  cd build
+  cmake -DCMAKE_PREFIX_PATH=/path/to/libtorch ..
+  make
+
+where ``/path/to/libtorch`` should be the full path to the unzipped LibTorch
+distribution. If all goes well, it should look something like this:
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app# mkdir build
+  root@4b5a67132e81:/example-app# cd build
+  root@4b5a67132e81:/example-app/build# cmake -DCMAKE_PREFIX_PATH=/path/to/libtorch ..
+  -- The C compiler identification is GNU 5.4.0
+  -- The CXX compiler identification is GNU 5.4.0
+  -- Check for working C compiler: /usr/bin/cc
+  -- Check for working C compiler: /usr/bin/cc -- works
+  -- Detecting C compiler ABI info
+  -- Detecting C compiler ABI info - done
+  -- Detecting C compile features
+  -- Detecting C compile features - done
+  -- Check for working CXX compiler: /usr/bin/c++
+  -- Check for working CXX compiler: /usr/bin/c++ -- works
+  -- Detecting CXX compiler ABI info
+  -- Detecting CXX compiler ABI info - done
+  -- Detecting CXX compile features
+  -- Detecting CXX compile features - done
+  -- Looking for pthread.h
+  -- Looking for pthread.h - found
+  -- Looking for pthread_create
+  -- Looking for pthread_create - not found
+  -- Looking for pthread_create in pthreads
+  -- Looking for pthread_create in pthreads - not found
+  -- Looking for pthread_create in pthread
+  -- Looking for pthread_create in pthread - found
+  -- Found Threads: TRUE
+  -- Configuring done
+  -- Generating done
+  -- Build files have been written to: /example-app/build
+  root@4b5a67132e81:/example-app/build# make
+  Scanning dependencies of target example-app
+  [ 50%] Building CXX object CMakeFiles/example-app.dir/example-app.cpp.o
+  [100%] Linking CXX executable example-app
+  [100%] Built target example-app
+
+If we supply the path to the serialized ``ResNet18`` model we created earlier
+to the resulting ``example-app`` binary, we should be rewarded with a friendly
+"ok":
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app/build# ./example-app model.pt
+  ok
+
+Step 4: Executing the Script Module in C++
+----------------------------------------
+
+Having succesfully loaded our serialized ``ResNet18`` in C++, we are now just a
+couple lines of code away from executing it! Let's add those lines to our C++
+application's ``main()`` function:
+
+.. code-block:: cpp
+
+    // Create a vector of inputs.
+    std::vector<torch::jit::IValue> inputs;
```

Comment:
I think IValue is being moved to ATen core, so this might need an update in the future

Reply:
We should keep a consistent alias to it at either `torch::IValue` or `torch::jit::IValue` just like we do for `torch::Tensor`